src/Collector/TemplateRenderCollector.php

```diff
@@ -0,0 +1,239 @@
+            $variables = $this->buildVariables($includeTemplateParamsArgument->value->left, $scope);
+        } else {
+            $variables = [];
+        }
+
+        return $this->buildTemplateRenders($node, $scope, $paths, $variables);
+    }
+
+    /**
+     * @param MethodCall $node
+     * @phpstan-return null|CollectedTemplateRenderArray[]
+     */
+    public function processNodeInPhp(Node $node, Scope $scope): ?array
+    {
+        $calledMethodName = $this->nameResolver->resolve($node->name);
+        if (!in_array($calledMethodName, ['render', 'renderToString'], true)) {
+            return null;
+        }
+
+        $calledType = $scope->getType($node->var);
+        if (!$this->templateTypeResolver->resolve($calledType)) {
+            return null;
+        }
+
+        $templatePathArg = $node->getArgs()[0] ?? null;
+        $tempalteVariablesArg = $node->getArgs()[1] ?? null;
+
+        if ($templatePathArg === null || $templatePathArg->value === null || $scope->getType($templatePathArg->value)->isNull()->yes()) {
+            $paths = [null]; // path not provided
+        } else {
+            $paths = $this->pathResolver->resolve($templatePathArg->value, $scope);
+        }
+
+        $variables = $this->buildVariables($tempalteVariablesArg->value ?? null, $scope);
+
+        return $this->buildTemplateRenders($node, $scope, $paths, $variables);
+    }
+
+    /**
+     * @param array<?string> $paths
+     * @param Variable[] $variables
+     * @return null|CollectedTemplateRenderArray[]
+     */
+    private function buildTemplateRenders(Node $node, Scope $scope, ?array $paths, array $variables): ?array
+    {
+        if ($paths === null) {
+            return [$this->buildTemplateRender($node, $scope, false, $variables)->toArray()];
+        }
+        $templateRenders = [];
+        foreach ($paths as $path) {
+            $templateRenders[] = $this->buildTemplateRender($node, $scope, $path, $variables)->toArray();
+        }
+        return count($templateRenders) > 0 ? $templateRenders : null;
+    }
+
+    /**
+     * @param false|string|null $path
+     * @param Variable[] $variables
+     */
+    private function buildTemplateRender(Node $node, Scope $scope, $path, array $variables): CollectedTemplateRender
+    {
+        return new CollectedTemplateRender(
+            $path,
+            $variables,
+            $scope->getClassReflection() !== null ? $scope->getClassReflection()->getName() : '',
+            $scope->getFunctionName() ?? '',
+            $scope->getFile(),
+            $node->getStartLine()
+        );
+    }
+
+    /**
+     * @return Variable[]
+     */
+    private function buildVariables(?Expr $argument, Scope $scope): array
+    {
+        if ($argument === null) {
+            return [];
+        }
+
+        $argumentType = $scope->getType($argument);
+
+        // todo support object (tempalte type)
+
+        $variables = [];
+        if ($argumentType instanceof ConstantArrayType) {
```

Comment:
Use PHPStan mechnism of detecting array shapes to support wider variaty of expressions (like array assigned to variable)